### PR TITLE
Update OSDK documentations

### PR DESCRIPTION
**Pull request by StevenJiang1110**

### CODE REVIEW

**Review by tatetian — COMMENTED** [review]
Please follow [semantic line breaks](https://sembr.org/) to reformat all Markdown files.

File: docs/src/osdk/guide/create-project.md [hunk 1]
```diff
@@ -1 +1,130 @@
...
+
+#[aster_main]
+fn kernel_main() -> ! {
+    println!("Hello world from guest kernel!");
+    abort();
+}
+
+#[cfg(ktest)]
+mod tests {
+    #[ktest]
```
[the remark this answers is not in the document]
Comment (StevenJiang1110) [reply docs/src/osdk/guide/create-project.md]: I remove the unit tests from kernel template but the unit tests in library template are reserved.

**Comment by StevenJiang1110** [comment]
I temporarily disable integration test in this PR. The ci fails on waiting for lock on package. I will look at the problem later.

截屏2024-03-01 12 05 29

**Comment by StevenJiang1110** [comment]
> The ci fails on waiting for lock on package.

The problem is that for a project created by `cargo osdk new`, it will depend on `aster_main` crate at the main branch, a specified commit. However, this PR modifies `aster_main`, the created project cannot depend on the new `aster_main`. 

So the two test cases can be added back once the PR is merged and then modify the aster_frame version of `cargo osdk new` to the newest commit.

File: docs/src/osdk/guide/create-project.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tatetian) [reply docs/src/osdk/guide/create-project.md]: Sure

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**